sktime/transformations/series/compose.py

```diff
@@ -96,3 +97,204 @@ def inverse_transform(self, Z, X=None):
+    >>> transformer = ColumnComposition(Detrender())
+    >>> y_hat = transformer.fit_transform(X)
+    """
+
+    _required_parameters = ["transformer"]
+
+    def __init__(self, transformer, columns="all"):
+        self.transformer = transformer
+        self.columns = columns
+        self.transformers_ = None
+        super(ColumnComposition, self).__init__()
+
+    def fit(self, Z, X=None):
+        """
+        Iterates over columns (series) and applies
+        the fit function of the transformer.
+
+        Parameters
+        ----------
+        Z : pd.Series, pd.DataFrame
+
+        Returns
+        -------
+        self : an instance of self
+
+        """
+        z = check_series(Z, allow_numpy=False)
+        self._is_fitted = False
+
+        # univariate case
+        if isinstance(z, pd.Series):
+            self.transformer.fit(z, X)
+            self._is_fitted = True
+            return self
+        # multivariate case
+        else:
+            if self.columns == "all":
+                self.columns = z.columns
+
+            # make sure z contains all columns that the user wants to transform
+            Z_wanted_keys = set(self.columns)
+            Z_new_keys = set(z.columns)
+            difference = Z_wanted_keys.difference(Z_new_keys)
+            if len(difference) != 0:
+                raise ValueError("Missing columns" + str(difference) + "in Z.")
+
+            self.transformers_ = {}
+            for colname in self.columns:
+                transformer = clone(self.transformer)
+                self.transformers_[colname] = transformer
+                self.transformers_[colname].fit(z[colname], X)
+            self._is_fitted = True
+            return self
+
+    def transform(self, Z, X=None):
+        """Transform data.
+        Returns a transformed version of Z by iterating over specified
+        columns and applying the univariate series transformer to them.
+
+        Parameters
+        ----------
+        Z : pd.Series, pd.DataFrame
+
+        Returns
+        -------
+        Z : pd.Series, pd.DataFrame
+            Transformed time series(es).
+        """
+        self.check_is_fitted()
+        z = check_series(Z)
+
+        # univariate case
+        if isinstance(z, pd.Series):
+            z = self.transformer.transform(z, X)
+            return z
+
+        # multivariate case
+        else:
+            # make copy of z
+            z = z.copy()
+            # make sure z contains all columns that the user wants to transform
+            z_wanted_keys = set(self.columns)
+            z_new_keys = set(z.columns)
+            difference = z_wanted_keys.difference(z_new_keys)
+            if len(difference) != 0:
+                raise ValueError("Missing columns" + str(difference) + "in Z.")
+            for colname in self.columns:
+                # self.columns : columns that are supposed to be transformed
+                self.transformers_[colname].check_is_fitted()
```

Comment:
This will happen inside the transform call in the next line and we have checked it at the beginning of this method
```suggestion
```